site/en/tutorials/load_data/images.ipynb

```diff
@@ -0,0 +1,1356 @@
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### A dataset of `(image, label)` pairs"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "YgvrWLKG67-x",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Using the same `from_tensor_slices` method we can build a dataset of labels"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "AgBsAiV06udj",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "label_ds = tf.data.Dataset.from_tensor_slices(tf.cast(all_image_labels, tf.int64))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "HEsk5nN0vyeX",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "for label in label_ds.take(10):\n",
+        "  print(label_names[label.numpy()])"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "jHjgrEeTxyYz",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Since the datasets are in the same order we can just zip them together to get a dataset of `(image, label)` pairs."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "AOEWNMdQwsbN",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "image_label_ds = tf.data.Dataset.zip((image_ds, label_ds))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "yA2F09SJLMuM",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The new dataset's `shapes` and `types`, are tuples of shapes and types as well, describing each field:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "DuVYNinrLL-N",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "print('image shape: ', image_label_ds.output_shapes[0])\n",
+        "print('label shape: ', image_label_ds.output_shapes[1])\n",
+        "print('types: ', image_label_ds.output_types)\n",
+        "print()\n",
+        "print(image_label_ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "vYGCgJuR_9Qp",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Basic methods for training"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "wwZavzgsIytz",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "To train a model with this dataset you will want the data:\n",
+        "\n",
+        "* To be well shuffeled.\n",
+        "* To be batched.\n",
+        "* To repeat forever.\n",
```

Comment:
This part is TBD with the Keras API. I think we're leaning towards *not* repeating the dataset itself forever, but instead iterating over the same finite dataset for a given number of epochs inside Keras.

Reply:
Acknowledged.

+1, that would be  a more intuitive API (I've already mentioned this to the keras team in b/109665994. Do you know the status of that plan?).

AFAIK, currently the only way it works is for the user to pass a `Dataset` that's long enough (typically using `.repeat()`) and use the `steps_per_epoch` argument to `fit`.

So I'll leave this for now, and gladly fix this when the behavior is changed.